.github/workflows/rebuilder.py

```diff
@@ -121,30 +177,28 @@ def get_monthly_rebuild_packages(pkgs: set[str], copr_pkgs: list[dict]) -> set[s
     >>> c = {"name" : "c", "builds" : { "latest" : { "id" : 2 } , "latest_succeeded" : { "id" : 1 } } }
     >>> d = {"name" : "d", "builds" : { "latest" : { "id" : 2 } , "latest_succeeded" : { "id" : 2 } } }
     >>> pkgs = { "b", "c", "d"}
-    >>> copr_pkgs = [a, b, c, d]
+    >>> copr_pkgs = [CoprPkg(p) for p in [a, b, c, d]]
     >>> rebuild_pkgs = get_monthly_rebuild_packages(pkgs, copr_pkgs)
     >>> print(rebuild_pkgs)
     {'d'}
     """
 
     for p in copr_pkgs:
-        latest_succeeded = p["builds"]["latest_succeeded"]
-        latest = p["builds"]["latest"]
-        if p["name"] not in pkgs:
+        if p.name not in pkgs:
             continue
-        if not latest_succeeded:
-            pkgs.discard(p["name"])
+        if not p.latest_succeeded:
+            pkgs.discard(p.name)
```

Comment:
Now that I read this code I wonder if all you needed to do was to change the type hint of the `copr_pkgs` argument. Well, it is a hint so it wouldn't be necessary but I know that the [`Munch`](https://github.com/Infinidat/munch) type returned from the Copr API behaves like a `dict` but you can access elements like in objects.

In other words: The `p.name` access is possible even without having `CoprPkg`.

If you still want `CoprPkg` which I think makes sense you can experiment with:

```python
class CoprPkg(Munch):
    pass
```

or if you want to really show what properties there are:

```python
class CoprPkg(Munch):
    @property
    def name(self) -> str:
        return self.name
```

The same applies to `CoprBuild`. Note that you may need to wrap `Munch` responses in your own classes to fully work here. But that's worth it, depending on how much your classes actually can do.